connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskMockitoTest.java

```diff
@@ -0,0 +1,763 @@
+        consumedOffsets.put(TOPIC_PARTITION3, new OffsetAndMetadata(FIRST_OFFSET + 30));
+        group1.recordConsumedOffsets(consumedOffsets);
+
+        assertEquals(0.333, metrics.currentMetricValueAsDouble(group.metricGroup(), "sink-record-read-rate"), 0.001d);
+        assertEquals(0.667, metrics.currentMetricValueAsDouble(group.metricGroup(), "sink-record-send-rate"), 0.001d);
+        assertEquals(9, metrics.currentMetricValueAsDouble(group.metricGroup(), "sink-record-active-count"), 0.001d);
+        assertEquals(4, metrics.currentMetricValueAsDouble(group.metricGroup(), "partition-count"), 0.001d);
+        assertEquals(5, metrics.currentMetricValueAsDouble(group.metricGroup(), "offset-commit-seq-no"), 0.001d);
+        assertEquals(3, metrics.currentMetricValueAsDouble(group.metricGroup(), "put-batch-max-time-ms"), 0.001d);
+
+        // Close the group
+        group.close();
+
+        for (MetricName metricName : group.metricGroup().metrics().metrics().keySet()) {
+            // Metrics for this group should no longer exist
+            assertFalse(group.metricGroup().groupId().includes(metricName));
+        }
+        // Sensors for this group should no longer exist
+        assertNull(group.metricGroup().metrics().getSensor("source-record-poll"));
+        assertNull(group.metricGroup().metrics().getSensor("source-record-write"));
+        assertNull(group.metricGroup().metrics().getSensor("poll-batch-time"));
+
+        assertEquals(0.667, metrics.currentMetricValueAsDouble(group1.metricGroup(), "sink-record-read-rate"), 0.001d);
+        assertEquals(1.333, metrics.currentMetricValueAsDouble(group1.metricGroup(), "sink-record-send-rate"), 0.001d);
+        assertEquals(45, metrics.currentMetricValueAsDouble(group1.metricGroup(), "sink-record-active-count"), 0.001d);
+        assertEquals(40, metrics.currentMetricValueAsDouble(group1.metricGroup(), "partition-count"), 0.001d);
+        assertEquals(50, metrics.currentMetricValueAsDouble(group1.metricGroup(), "offset-commit-seq-no"), 0.001d);
+        assertEquals(30, metrics.currentMetricValueAsDouble(group1.metricGroup(), "put-batch-max-time-ms"), 0.001d);
+    }
+
+    private void expectRebalanceLossError(RuntimeException e) {
+        doThrow(e).when(sinkTask).close(INITIAL_ASSIGNMENT);
+    }
+
+    private void expectRebalanceRevocationError(RuntimeException e) {
+        doThrow(e).when(sinkTask).close(INITIAL_ASSIGNMENT);
+        when(sinkTask.preCommit(anyMap())).thenReturn(Collections.emptyMap());
+    }
+
+    private void expectRebalanceAssignmentError(RuntimeException e) {
+        when(sinkTask.preCommit(anyMap())).thenReturn(Collections.emptyMap());
+        when(consumer.position(TOPIC_PARTITION)).thenReturn(FIRST_OFFSET);
+        when(consumer.position(TOPIC_PARTITION2)).thenReturn(FIRST_OFFSET);
+
+        doThrow(e).when(sinkTask).open(INITIAL_ASSIGNMENT);
+    }
+
+    private void verifyInitializeTask() {
```

Comment:
IMO we should also stick to the pattern of capturing the rebalance listener when it's passed to our mocked consumer's `subscribe` method, since this doesn't require a testing-only method (i.e., `getRebalanceListener`) to be added to the `WorkerSinkTask` class.

Reply:
I'll revert this change, but admittedly am not a fan of the current mechanism for capturing the listener - I feel it's too tightly coupled to the current implementation. 

There's no actual benefit on having a new `HandleRebalance()` instance being created each `#initializeAndStart()` invocation vs having it as a final member in the class.

Reply:
> There's no actual benefit on having a new HandleRebalance() instance being created each #initializeAndStart() invocation vs having it as a final member in the class.

Fair--I'm actually fine with keeping it as a final member; I'm more concerned with not exposing it via a getter method and, instead, capturing it by recording calls to a (mocked) consumer's `subscribe` method. Thoughts?

Reply:
> > There's no actual benefit on having a new HandleRebalance() instance being created each #initializeAndStart() invocation vs having it as a final member in the class.
> 
> Fair--I'm actually fine with keeping it as a final member; I'm more concerned with not exposing it via a getter method and, instead, capturing it by recording calls to a (mocked) consumer's `subscribe` method. Thoughts?

Ok, I'll do that instead